src/shared_types.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+use crate::{utils, PublicKey, Result, XorName};
+use multibase::Decodable;
+use serde::{Deserialize, Serialize};
+use std::ops::Range;
+
+pub type Key = Vec<u8>;
+pub type Value = Vec<u8>;
+pub type KvPair = (Key, Value);
```

Comment:
Yep, it is used. 
About the name, yeah `Kv` is not so nice. I did use `KeyValue` here at some point as well and I generally don't like abbreviations. But anyway, I think `KeyValuePair` is most expressive, since it `KeyValue` could be meaning "the value of a key" and isn't necessarily "the key and its value".

Reply:
Agreed, I like `KeyValuePair`!